lib/timeout.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+// Default timeout to 10 minutes if not provided
+const kDefaultTimeout = 1000 * 60 * 10;
+
+function timeout(context, proc, next, step) {
+  let hasRun = false;
+  const delay = context.options.timeoutLength || kDefaultTimeout;
+  const timeout = setTimeout(finish, delay, null, null, true);
```

Comment:
that's that way I overload `finish` to know it was a timeout (third arg `true`) otherwise it acts like a "regular" callback `(err, ret)`.
The `if` in L19 will overwrite `err` and `ret` with info about timing out anyway, so first 2 args are ignored.